internal/cmd/auth.go

```diff
@@ -164,7 +166,13 @@ func authenticationGRPC(
 				rgxs = append(rgxs, rgx)
 			}
 
-			interceptors = append(interceptors, auth.EmailMatchingInterceptor(logger, rgxs))
+			interceptors = append(interceptors, auth.EmailMatchingInterceptor(logger, rgxs, authOpts...))
+		}
+
+		if authCfg.Methods.Token.Enabled {
+			// dont require namespace auth for interacting with auth service
+			authOpts = append(authOpts, auth.WithServerSkipsAuthentication(authServer))
+			interceptors = append(interceptors, auth.NamespaceMatchingInterceptor(logger, authOpts...))
```

Comment:
We need to change this from skipping the namespace check, to enforcing that we always reject requests for auth with namespaced tokens.

Otherwise, this opens up a security hole where you can use a namespaced token to create non-namespaced one and then accessing everything.